authors/apps/authentication/serializers.py

```diff
@@ -15,6 +17,16 @@ class RegistrationSerializer(serializers.ModelSerializer):
         min_length=8,
         write_only=True
     )
+    def validate(self, data):
+        username = data.get('username', None)
+        email = data.get('email', None)
+        password = data.get("password", None)
+
+        validator = ValidateRegistrationData()
+        validator.password_validation(password)
+        validator.username_validation(username)
+        return {'email': email, 'username': username, 'password': password}
```

Comment:
it would be better for you to first assign the dictionary your trying to return in a variable such as a user then return that variable

Reply:
It is a simple straightforward dictionary. It is not complex and your suggestion adds more lines of code. Im thinking we should leave it as is